indico/modules/events/contributions/controllers/display.py

```diff
@@ -97,9 +98,21 @@ def _check_access(self):
             raise Forbidden
 
     def _process(self):
-        contributions = get_contributions_with_user_as_submitter(self.event, session.user)
+        contribs = get_contributions_for_user(self.event, session.user)
+        categorized = defaultdict(set)
+        for contrib in contribs:
+            for link in contrib.person_links:
+                if link.is_speaker:
+                    categorized['speaker'].add(contrib)
+                if link.author_type == AuthorType.primary:
+                    categorized['primary'].add(contrib)
+                if link.author_type == AuthorType.secondary:
+                    categorized['secondary'].add(contrib)
+            for principal in contrib.acl_entries:
+                if principal.user == session.user and principal.has_management_permission('submit'):
```

Comment:
This could be simplified to `if contrib.can_manage(session.user, 'submit', allow_admin=False, check_parent=False)`